src/libs/jira-lib.ts

```diff
@@ -0,0 +1,339 @@
+import * as dotenv from "dotenv";
+import axios, { AxiosInstance } from "axios";
+
+dotenv.config();
+
+export interface Issue {
+  [name: string]: any;
+}
+
+export class Jira {
+  private isDryRun: boolean;
+  private dryRunIssueCounter: number = 0;
+  private axiosInstance: AxiosInstance;
+  jiraClosedStatuses: string[];
+
+  constructor() {
+    Jira.checkEnvVars();
+
+    // Interpret DRY_RUN environment variable flexibly
+    this.isDryRun = process.env.DRY_RUN?.trim().toLowerCase() === 'true';
+
+    this.axiosInstance = axios.create({
+      baseURL: process.env.JIRA_BASE_URI,
+      headers: {
+        "Authorization": `Bearer ${process.env.JIRA_TOKEN}`,
+        "Content-Type": "application/json",
+      },
+    });
+    this.jiraClosedStatuses = process.env.JIRA_CLOSED_STATUSES
+      ? process.env.JIRA_CLOSED_STATUSES.split(",").map((status) =>
+          status.trim()
+        )
+      : ["Done"];
+  }
+  async getCurrentUser() {
+    try {
+      const response = await this.axiosInstance.get('/rest/api/2/myself');
+      return response.data;
+    } catch (error) {
+      throw new Error(`Error fetching current user: ${error}`);
+    }
+  }
+  async getIssueTransitions(issueId: string) {
+    try {
+      const response = await this.axiosInstance.get(`/rest/api/2/issue/${issueId}/transitions`);  
+      return response.data.transitions;
+    } catch (error) {
+      throw new Error(`Error fetching issue transitions: ${error}`);
+    }
+  }
+  async transitionIssue(issueId: string, transitionData: any) {
+    if (this.isDryRun) {
+      console.log(`[Dry Run] Would transition issue ${issueId} with data:`, transitionData);
+      return;
+    }
+
+    try {
+      const response = await this.axiosInstance.post(`/rest/api/2/issue/${issueId}/transitions`, transitionData);
+      console.log(`Issue ${issueId} transitioned successfully.`);
+    } catch (error) {
+      throw new Error(`Error transitioning issue ${issueId}: ${error}`);
+    }
+  }
+  async getPriorities() {
+    try {
+      const response = await this.axiosInstance.get('/rest/api/2/priority');
+      return response.data;
+    } catch (error) {
+      throw new Error(`Error fetching priorities: ${error}`);
+    }
+  }
+  async getIssue(issueId: string) {
+    try {
+      const response = await this.axiosInstance.get(`/rest/api/2/issue/${issueId}`);
+      return response.data;
+    } catch (error) {
+      throw new Error(`Error transitioning issue ${issueId}: ${error}`);
+    }
+  }
+  async removeCurrentUserAsWatcher(issueId: string) {
+    try {
+      const currentUser = await this.getCurrentUser();
+      console.log("Remove watcher: " + currentUser.name)
+
+      if (this.isDryRun) {
+        console.log(`[Dry Run] Would remove ${currentUser.name} from ${issueId} as watcher.`);
+        return; // Skip the actual API call
+      }
+
+      await this.axiosInstance.delete(`/rest/api/2/issue/${issueId}/watchers`, {
+        params: {
```

Comment:
Did you confirm that this delete request works with the `params` object? When I made my [PR](https://github.com/Enterprise-CMCS/macpro-security-hub-sync/blob/1205dd9501938029b18837795b3a0531ad3d6a3e/src/libs/jira-lib.ts#L84), I had to explicitly set `username` as a query parameter in the URL. Just want to be sure it still works with this syntax.

Reply:
Hmm, not sure what issue you ran into but I verified this works fine at least against my local container jira.

Reply:
Also works against my local jira; seems like this syntax works just fine.